src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,677 @@
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="ushort"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        [CLSCompliant(false)]
+        public bool TryGetUInt16(out ushort value) => ushort.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="uint"/> equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="uint"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        [CLSCompliant(false)]
+        public bool TryGetUInt32(out uint value) => uint.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="ulong"/> equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="ulong"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        [CLSCompliant(false)]
+        public bool TryGetUInt64(out ulong value) => ulong.TryParse(_value, out value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="decimal"/> equivalent.
+        ///   A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="decimal"/> value equivalent of the number contained in this instance,
+        ///   if the conversion succeeded, or zero if the conversion failed.
+        /// </param>
+        /// <returns>
+        ///  <see langword="true"/> if instance was converted successfully; 
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        public bool TryGetDecimal(out decimal value) => decimal.TryParse(_value, NumberStyles.Float, CultureInfo.InvariantCulture, out value);
```

Comment:
I think using float in this case is incorrect. I wouldn't use any NumberStyle, or use the default for decimal, which is NumberSytles.Number (which includes allowtrailingsign, and allowthousands, but doesn't allowexponents).

Same with GetDecimal.

https://github.com/dotnet/corefx/blob/3cdd1f49a13e61060e24c8221ec381ce4cc3fc94/src/Common/src/CoreLib/System/Decimal.cs#L464-L468
https://github.com/dotnet/corefx/blob/3cdd1f49a13e61060e24c8221ec381ce4cc3fc94/src/Common/src/CoreLib/System/Globalization/NumberStyles.cs#L58-L59

Maybe add a test too.
```C#
Console.WriteLine(decimal.Parse("1,000"));
Console.WriteLine(decimal.Parse("1,000", NumberStyles.Number));
Console.WriteLine(decimal.Parse("1,000", NumberStyles.Float)); // Throws System.FormatException: Input string was not in a correct format.
```

Reply:
I am afraid with `TryParse` I have to specify `NumberStyle`, but I will change it into `NumberStyle.Number` for decimal and add tests.